slash/plugins/__init__.py

```diff
@@ -119,8 +127,15 @@ def install(self, plugin, activate=False, activate_later=False, is_internal=Fals
         if not isinstance(plugin, PluginInterface):
             raise IncompatiblePlugin("Invalid plugin type: {0!r}".format(type(plugin)))
         plugin_name = plugin.get_name()
+        if re.search(r'[^A-Za-z0-9_\s-]', plugin_name):
+            raise IllegalPluginName("Illegal plugin name: {}".format(plugin_name))
+
+        if any(char in plugin_name for char in _DEPRECATED_CHARACTERS):
+            warn_deprecation("Unsupported characters in plugin name: '{}'".format(plugin_name))
```

Comment:
Deprecation should explain the full reason - "in the future, dashes and underscore will not be allowed in plugin names - spaces should be used instead"